File: accommodation-service/model/accommodation.go
package model

import "github.com/google/uuid"

type Accommodation struct {
	ID                   uuid.UUID
	HostId               uuid.UUID
	Name                 string
	Benefits             string
	MinGuests            int
	MaxGuests            int
	AutomaticReservation int
	BasePrice            float64
	Location             string
}

File: public-app/src/shared/model/accommodation.ts
import { Discount, Period } from ".";
import { AccommodationRating } from "./rating";

export interface Accommodation {
  id?: string;
  name?: string;
  benefits?: string;
  minGuests?: number;
  maxGuests?: number;
  basePrice?: number;
  location?: string;
  automaticReservation?: number;
  periods?: Period[];
  discounts?: Discount[];
  ratings?: AccommodationRating[];
}